### ISSUE TITLE
Specify dependency path of packages that should be bumped

### ISSUE THREAD

**Issue by aspirisen** [atlassian/changesets#512]
## Problem

I have i.e. four packages: `app`, `page-users`, `page-settings`, `ui-kit`. 
"app" is using "page-users" and "page-settings"
"page-users" and "page-settings" are using "ui-kit"

```
                                
              app              
             /    \             
            /      \            
           /        \           
          -          -          
     page-users    page-settings
          \          /          
           \        /           
            \      /            
             -    -           
             ui-kit          
                           
```
For example if i make a breaking change in `ui-kit` and use it in `page-users` i need to bump all the packages except `page-settings`, it will use the old version of `ui-kit`, by that i will know that my changes will not break `page-settings`.

I need to specify a tree path how `changesets` will bump the versions, maybe like `linked-packages`

## Proposed solution

Probably my use case is very specific, maybe it would be possible to overload the way how `changesets` determines what packages should be bumped.

**Comment by Andarist** [comment]
Could you describe this in more detail? I fail to see why `page-settings` shouldn't be bumped if it depends on `ui-kit` that has been changed.

**Comment by aspirisen** [comment]
@Andarist to make the application more tough, if somebody will change `ui-kit` and make some bug it will not affect `page-settings` as it will point to previous version of `ui-kit`. So there will be no accidental issues across all application. 

Just imagine a very large application with dozens of teams that are working on the same code base. Some pages are not touched for months since some work on them has been finished. And if some of the developers will break ui-kit it will affect all the pages, but because of the large application this change can be missed on this page because it is "canned". Yes, this will increase the size of the output bundle as there will be two versions of the same package, but it is trade of to increase stability and make the app more bullet proof.

**Comment by Andarist** [comment]
Seems like this PR should help you achieve what u want: https://github.com/atlassian/changesets/pull/507

It might still require some extra scripting on your part though that would switch away from workspace protocol for certain pkgs

**Comment by aspirisen** [comment]
@Andarist from first look seems like what I need, thanks 👍

**Comment by aspirisen** [comment]
Close in favor of https://github.com/atlassian/changesets/issues/506

**Comment by aspirisen** [comment]
Looks like this case https://github.com/atlassian/changesets/pull/507 is not exactly the same. 
@Andarist is it possible to specify custom bump tree resolution? If no, does it make sense to introduce it, that will help to cover edge cases.

**Comment by Andarist** [comment]
I understand that the issue there is not exactly the same as yours. It should be possible to switch between workspace protocol and regular dependency ranges using a custom script - prior to running Changesets, so it could skip bumping some of the dependencies that would lack the workspace protocol, right?

**Comment by aspirisen** [comment]
@Andarist yes, as a workaround, also in this script i have to detect the changes packages comparing with main branch.
I created a PR to override logic for bumping versions, so now you can handle any edge case.
